### ISSUE TITLE
[Linux] Minimize to tray (instead of closing the app)

### ISSUE THREAD

**Issue by Fincer** [EragonJ/Kaku#528]
#### Bug

The application has a tray icon. As an user clicks "Close" button on the application main window, the application closes. I would prefer minimizing it to the tray and closing it by selecting such option after right-clicking the tray icon.

May need an additional application menubar with File -> Close option?

#### Steps to reproduce the bug

Press hard-coded "close" button. Expected the application to be minimized in the tray. Instead, the whole app closes.

#### What is your Kaku version?

1.9.0

#### What is your OS (with version number)?

Arch Linux x86_64 (KDE/Plasma 5 desktop)

**Comment by EragonJ** [comment]
@Fincer @funilrys is that the native behavior in linux ?

**Comment by Fincer** [comment]
Minimizing to tray by pressing a close button is native or achievable behavior for these apps, at least:
- Valve Steam Client
- Thunderbird email
- Pidgin messenger
- VLC media player?
- Whatsapp desktop (it's an electron app by the way, hosted on [Whatsapp-Desktop/Github](https://github.com/Enrico204/Whatsapp-Desktop))

Personally, I'd pretty much prefer the behavior which is implemented in Whatsapp-desktop, for example. Kaku has a tray icon but it's not really that useful because the main app window gets just minimized to the main application panel, too. Minimizing Kaku only to tray area is impossible with the current tray implementation.

Here's a screenshot illustrating the current situation (please pay attention to Whatsapp and Kaku): [Screenshot - Kaku & Whatsapp tray](https://i.imgur.com/GxnO4be.png)

Both Whatsapp and Kaku are minimized the same way but only Kaku is presented in the application panel, too.

**Comment by EragonJ** [comment]
Because I don't have Linux, what I can do is quickly scan whatsapp desktop's source code, so if I misunderstood things wrong, please tell me.

https://github.com/Enrico204/Whatsapp-Desktop/blob/master/app/main.js#L532-L539

Based on these lines, it looks like how they `show/hide window` and `quit app` is based on whether you did enable `tray` settings or not https://github.com/Enrico204/Whatsapp-Desktop/blob/master/app/js/views/settings.js#L91

If `tray` settings is not enabled when you close the window, the app will be quitted directly. Instead, if it's enabled, the window will be hidden.

If this is the native way in Linux, we can follow this way if needed. This can be an improvement issue.

**Comment by Fincer** [comment]
Yes, it is exactly as you described it. "Show Tray Icon" is toggleable option in Whatsapp-Desktop settings and behaves as you said.

I'd absolutely prefer that kind of behavior for Kaku, too. Implementing "Show in tray" toggle button in Kaku Settings and similar code part would be good enough for Linux at least. :+1:

**Comment by p2635** [comment]
Same for Windows 10 please.
